composeApp/src/commonMain/kotlin/sp/bvantur/tasky/home/presentation/HomeViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package sp.bvantur.tasky.home.presentation
+
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+import sp.bvantur.tasky.core.presentation.ViewStateViewModel
+import sp.bvantur.tasky.home.domain.HomeRepository
+
+class HomeViewModel(private val repository: HomeRepository) : ViewStateViewModel<HomeViewState>(HomeViewState()) {
+
+    override suspend fun initialLoadData() {
+        super.initialLoadData()
+        repository.getTodayAgenda(Clock.System.now().toEpochMilliseconds())
```

Comment:
Where do you usually put such "static" values? It is always the same code call so it could also be called in data source.

Reply:
Could you elaborate on what you mean? Wouldn’t you want to call the `getTodayAgenda()` each time the user selects a new day to view?

Reply:
Yeah, my choice of words wasn't the best here. I am talking about the next part of the code:
`Clock.System.now().toEpochMilliseconds()`

This line will always be called and it won't change. Would it make more sense to move it to the data layer directly when we call remote API, or should I keep the flexibility I have now by leaving the possibility to pass any timestamp to the repository to fetch?

Reply:
You are correct, if this function would always fetch the agenda for today then remove the `time` parameter and call `Clock.System.now().toEpochMilliseconds()` in the data layer.

But I think your function’s name is a bit misleading. The API takes the timestamp value to give you the agenda for that particular day. Otherwise, it would not have the `time` parameter and just provide you with today’s agenda. 

What I meant in my original response is that when the user taps on a different date on the Agenda screen, it would be a good time to call this function of yours and pass it the date the user selected. Remember that the `/agenda` endpoint only returns the agenda items for a particular day, not all agenda items the user has.

I suggest you rename your function to something that indicates that it will fetch all the agenda items for a day provided by the `time` parameter and couple this function call with the user changing their selected date on the Agenda screen.